src/stactools/core/utils/convert.py

```diff
@@ -16,25 +17,102 @@
             used.
     """
-    if not utils.gdal_driver_is_enabled("COG"):
-        raise DriverRegistrationError(
-            "GDAL's COG driver is not enabled, make sure you're using GDAL >= 3.1"
-        )
-    destination_profile = DEFAULT_PROFILE.copy()
+    check_gdal_driver()
+
+    src = rasterio.open(infile)
+    dest_profile = DEFAULT_PROFILE.copy()
+    dest_profile.update(
+        {
+            "dtype": rasterio.uint8,
+            "width": src.width,
+            "height": src.height,
+            "crs": src.crs,
+            "transform": src.transform,
+        }
+    )
+
     if profile:
-        destination_profile.update(profile)
+        dest_profile.update(profile)
+
+    # If a band number was provided, create a single-band COG
+    if band:
+        single_band = src.read(band)
+        dest_profile.update({"count": 1})
+        with rasterio.open(outfile, "w", **dest_profile) as dest:
+            dest.write(single_band, 1)
+    # If no band numbers were provided, create a multi-band COG
+    else:
+        dest_profile.update({"count": src.count})
+        rasterio.shutil.copy(infile, outfile, **dest_profile)
+
+
+def cogify_subdataset(infile: str, outfile: str) -> None:
+    """Exports a Cloud-Optimized GeoTIFF (COG) from a subdataset within an HDF file.
+    Uses :py:meth:`rasterio.shutil.copy`.
+    Args:
+        infile (str): The input file.
+        outfile (str): The output COG to be written.
+    """
+    check_gdal_driver()
+    destination_profile = DEFAULT_PROFILE.copy()
     rasterio.shutil.copy(infile, outfile, **destination_profile)
+
+
+def list_subdataset(infile: str, outdir: str) -> Tuple[List[str], List[str]]:
+    """Generates lists of output paths and subdataset names for COGs created from an HDF file.
+    This is then used to call the cogify_subdataset() function to export out COG files.
+    Args:
+        infile (str): The input HDF file
+        outdir (str): The output directory where the HDF files will be created
+    Returns:
+        Tuple[List[str], List[str]]: A two tuple (paths, names):
+            - The first element is a list of the output tiff paths
+            - The second element is a list of subdataset names
+    """
+
+    with rasterio.open(infile) as dataset:
+        subdatasets = cast(List[str], dataset.subdatasets)
+        base_file_name = os.path.splitext(os.path.basename(infile))[0]
+        paths = []
+        subdataset_names = []
+        for subdataset in subdatasets:
+            parts = subdataset.split(":")
+            subdataset_name = parts[-1]
+            sanitized_subdataset_name = subdataset_name.replace(" ", "_").replace(
+                "/", "_"
+            )
+            subdataset_names.append(sanitized_subdataset_name)
+            file_name = f"{base_file_name}_{sanitized_subdataset_name}.tif"
+            outfile = os.path.join(outdir, file_name)
+            cogify_subdataset(subdataset, outfile)
+            paths.append(outfile)
```

Comment:
We should add some logic to make sure the subdataset can be a valid raster -- e.g. sometimes there are single variable subdatasets that don't make sense to convert. See https://github.com/stactools-packages/viirs/blob/ad062e9d84dcfef05009f78453b2c6fa849fdab7/src/stactools/viirs/cog.py#L56-L61 for an example of this type of check.